### PR TITLE
manualintegrate modifications

### LINKED ISSUE #9687: integrate(1/((x**2+4)*sqrt(4*x**2+1)), x) raises `TypeError: 'NoneType' object has no attribute '__getitem__'`
```
In [1]: integrate(1/((x**2+4)*sqrt(4*x**2+1)), x)
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-1-b25b3537657d> in <module>()
----> 1 integrate(1/((x**2+4)*sqrt(4*x**2+1)), x)

/home/ondrej/repos/sympy/sympy/utilities/decorator.py in threaded_func(expr, *args, **kwargs)
     35                                       func(expr.rhs, *args, **kwargs))
     36             else:
---> 37                 return func(expr, *args, **kwargs)
     38 
     39     return threaded_func

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in integrate(*args, **kwargs)
   1253     if isinstance(integral, Integral):
   1254         return integral.doit(deep=False, meijerg=meijerg, conds=conds,
-> 1255                              risch=risch, manual=manual)
   1256     else:
   1257         return integral

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in doit(self, **hints)
    480                     function, xab[0],
    481                     meijerg=meijerg1, risch=risch, manual=manual,
--> 482                     conds=conds)
    483                 if antideriv is None and meijerg1 is True:
    484                     ret = try_meijerg(function, xab)

/home/ondrej/repos/sympy/sympy/integrals/integrals.pyc in _eval_integral(self, f, x, meijerg, risch, manual, conds)
    884             if h is None and manual is not False:
    885                 try:
--> 886                     result = manualintegrate(g, x)
    887                     if result is not None and not isinstance(result, Integral):
    888                         if result.has(Integral):

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in manualintegrate(f, var)
   1143     sympy.integrals.integrals.Integral
   1144     """
-> 1145     return _manualintegrate(integral_steps(f, var))

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in integral_steps(integrand, symbol, **options)
    944             null_safe(trig_substitution_rule)
    945         ),
--> 946         fallback_rule)(integral)
    947     del _integral_cache[cachekey]
    948     return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in do_one_rl(expr)
     83     def do_one_rl(expr):
     84         for rl in rules:
---> 85             result = rl(expr)
     86             if result != expr:
     87                 return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in do_one_rl(expr)
     83     def do_one_rl(expr):
     84         for rl in rules:
---> 85             result = rl(expr)
     86             if result != expr:
     87                 return result

/home/ondrej/repos/sympy/sympy/strategies/core.pyc in null_safe_rl(expr)
     63     """ Return original expr if rule returns None """
     64     def null_safe_rl(expr):
---> 65         result = rule(expr)
     66         if result is None:
     67             return expr

/home/ondrej/repos/sympy/sympy/integrals/manualintegrate.py in trig_substitution_rule(integral)
    688         for expr in matches:
    689             match = expr.match(a + b*symbol**2)
--> 690             a = match[a]
    691             b = match[b]
    692 

TypeError: 'NoneType' object has no attribute '__getitem__'
```

### PR DESCRIPTION
**Pull request by smichr**
fixes #9687
closes #10611 as a replacement and modification

### CODE REVIEW

**Comment by smichr** [comment]
ping @gyxd

**Comment by smichr** [comment]
ping @gxyd

File: sympy/solvers/tests/test_ode.py [hunk 1]
```diff
@@ -542,21 +542,17 @@ def test_checksysodesol():
...
-    "sqrt(5*_y**2 + 4*C1 - C2)), (_y, z(t))), C3 + Integral(sqrt(3)/6, t))]"
-    assert str(dsolve(eq1)) == sol1
-
-    eq2 = (4*diff(x(t),t) + 2*y(t)*z(t)*sin(t), 3*diff(y(t),t) - z(t)*x(t)*sin(t), 5*diff(z(t),t) - x(t)*y(t)*sin(t))
-    sol2 = "[Eq(Integral(3/(sqrt(-6*_y**2 - C1 + 5*C2)*sqrt(3*_y**2 + C1 - 4*C2)), (_y, x(t))), C3 + "\
-    "Integral(-sqrt(5)*sin(t)/10, t)), Eq(Integral(4/(sqrt(-4*_y**2 - 3*C1 + C2)*sqrt(-4*_y**2 + 5*C1 - C2)), "\
-    "(_y, y(t))), C3 + Integral(sqrt(15)*sin(t)/15, t)), Eq(Integral(5/(sqrt(-10*_y**2 - 3*C1 + C2)*"\
-    "sqrt(5*_y**2 + 4*C1 - C2)), (_y, z(t))), C3 + Integral(-sqrt(3)*sin(t)/6, t))]"
-    assert str(dsolve(eq2)) == sol2
+    eq1 = (4*diff(x(t),t) + 2*y(t)*z(t), 3*diff(y(t),t)
```
Comment (smichr) [inline sympy/solvers/tests/test_ode.py]: This doesn't satisfy the original ode, eq1, so it must be wrong.

**Comment by gxyd** [comment]
Sorry, I was (am) busy with other stuff today. I'll leave comments if I have any.

**Pull request merged**